src/contract/ga/index.ts

```diff
@@ -0,0 +1,298 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Generalized Account module - routines to use generalized account
+ *
+ * @module @aeternity/aepp-sdk/es/contract/ga
+ * @export GeneralizedAccount
+ * @example import { GeneralizedAccount } from '@aeternity/aepp-sdk'
+ */
+// @ts-expect-error TODO: remove me
+import Ae from '../../ae'
+import Contract, { _Contract } from '../../ae/contract'
+// @ts-expect-error TODO: remove me
+import { TX_TYPE } from '../../tx/builder/schema'
+// @ts-expect-error TODO: remove me
+import { buildTx, unpackTx } from '../../tx/builder'
+import { prepareGaParams } from './helpers'
+import { hash } from '../../utils/crypto'
+import { decode, EncodedData } from '../../utils/encoder'
+import { IllegalArgumentError, MissingParamError } from '../../utils/errors'
+
+export interface AuthData {
+  gasLimit: number
+  callData: string
+  source?: string
+  args?: object
+}
+
+interface GAObject {
+  owner: string
+  transaction: string
+  rawTx: string
+  gaContractId: string
+}
+
+interface Account {
+  authFun: string
+  balance: string
+  contractId?: string
+  id?: string
+  kind: string
+  nonce: number
+  payable: true
+}
+
+export interface IGeneralizedAccount {
+  readonly createGeneralizedAccount: (
+    authFnName: string,
+    source: string,
+    args?: never[],
+    options?: object
+  ) => Promise<Readonly<GAObject>>
+
+  readonly createMetaTx: (
+    rawTransaction: string,
+    authData: AuthData,
+    authFnName: string,
+    options?: object
+  ) => Promise<string>
+
+  readonly isGA: (address: string) => Promise<boolean>
+
+  readonly buildAuthTxHash: (
+    transaction: string,
+    options: {innerTx?: boolean, networkId?: string}
+  ) => Uint8Array
+}
+
+export abstract class _GeneralizedAccount extends _Contract implements IGeneralizedAccount {
```

Comment:
I believe it'd be if we classes didn't implement any interfaces. This way we won't have a need to assign values through `Parameters` and duplicate signatures. Classes can be assigned on values as types just like interfaces. (this is an opinion, proceed only if we're on the same page!)